README.md
# Roulette

## Gameplay
- to be added soon

## Bet types
### Outside Bets - 2 to 1 payouts
- 1st column, 2nd column, 3rd column
- 1st 12, 2nd 12, 3rd 12
### Outside Bets - 1 to 1 payouts
- red, black
- even, odd
- low (1 through 18), high (19 through 36) 

### Inside Bets
- Straight up (35 to 1 payout) - any single number, including 0  

### Inside Bets
###### _ (to be added in a later version) _
- Split (17 to 1 payout) - any two adjoining numbers vertical or horizontal
- Basket (11 to 1 payout) - 
- Street (11 to 1 payout) - any three numbers horizontal (1, 2, 3, or 4, 5, 6 etc.)
- Corner (8 to 1 payout) - any four adjoining numbers in a block (eg 17, 18, 20, 21)
- Six Line (5 to 1) - any six numbers from two rows (eg 28, 29, 30, 31, 32, 33)



changelog.md
# Changelog


This is my first time creating and writing a change-log. I'm planning on trying it out to see if it'll help me focus on being more efficient with my progress.

## 2021-02-04
- Considering completely rewriting script.js.

board.js
// board.js

// American Roulette Wheel
export const wheel = [
    "0", "28", "9", "26", "30", "11", "7", "20", "32", "17", "5", "22", "34",
    "15", "3", "24", "36", "13", "1", "00", "27", "10", "25", "29", "12", "8",
    "19", "31", "18", "6", "21", "33", "16", "4", "23", "35", "14", "2"
]

/**
 * Inside Bets
 * - Straight up (35 to 1 payout) - any single number, including 0
 * - Split (17 to 1 payout) - any two adjoining numbers vertical or horizontal
 * - Basket (11 to 1 payout) - 
 * - Street (11 to 1 payout) - any three numbers horizontal (1, 2, 3, or 4, 5, 6 etc.)
 * - Corner (8 to 1 payout) - any four adjoining numbers in a block (eg 17, 18, 20, 21)
 * - Six Line (5 to 1) - any six numbers from two rows (eg 28, 29, 30, 31, 32, 33)
 */

/**
 * Outside Bets - 2 to 1 payouts
 * - column1, column2, column3
 * - dozen1, dozen2, dozen3
 * 
 * Outside Bets - 1 to 1 payouts
 * - red, black
 * - even, odd
 * - low (1 through 18), high (19 through 36) 
 */
export const column1 = ["1", "4", "7", "10", "13", "16", "19", "22", "25", "28", "31", "34"]
export const column2 = ["2", "5", "8", "11", "14", "17", "20", "23", "26", "29", "32", "35"]
export const column3 = ["3", "6", "9", "12", "15", "18", "21", "24", "27", "30", "33", "36"]
export const green = ["0", "00"]

export const dozen1 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
export const dozen2 = [13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
export const dozen3 = [25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36]

export const red = ["1", "3", "5", "7", "9", "12", "14", "16", "18", "19", "21", "23", "25", "27", "30", "32", "34", "36"]
export const black = ["2", "4", "6", "8", "10", "11", "13", "15", "17", "20", "22", "24", "26", "28", "29", "31", "33", "35"]

export const even = [2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36]
export const odd = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35]

export const low = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]
export const high = [19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36]





script.js
import * as table from "/table.js"

// Variables from DOM
var wheelResult_span = document.querySelector("#wheelResult");
var prevResults_span = document.querySelector("#previousResults");
// TODO:  Add betAmount selections
var betSelections = Array.from(document.getElementsByClassName("selection"))


var displayWindow_p = document.querySelector("#displayWindow")
var messageDisplay_p = document.querySelector("#messageDisplay");
var betDisplay_p = document.querySelector("#betDisplay")
var bankBalance_span = document.querySelector("#bankBalance");
var betAmount = document.querySelector("#betAmount");
var betBtn = document.querySelector("#betBtn");
var spinBtn = document.querySelector("#spinBtn");

spinBtn.addEventListener("click", spinWheel);
betBtn.addEventListener("click", setBetAmount);
betSelections.forEach((selection) => selection.addEventListener("click", selectBetPlacement))

var bankBalance = 1000;
var wheelResult;
var bettingAllowed = true;
var betSelection

// Select betAmount
function selectBetPlacement(e) {
    betSelection = e.target.id
    // Display bet selection
    betDisplay_p.innerHTML = `Selected ${betSelection}`
    console.log(`Selected: ${betSelection}`)
}


// Select bet amount
function setBetAmount() {
    if (bettingAllowed && betSelection) {
        betDisplay_p.innerHTML = `Bet:  ${betAmount.value} on ${betSelection}`
        bankBalance -= betAmount.value;
        console.log(`Bet:  ${betAmount.value} on ${betSelection}`)
    } else {
        bettingAllowed
            ? betDisplay_p.innerHTML = "You must select a bet placement."
            : betDisplay_p.innerHTML = "No betting alllowed."
        console.log("Attempted to bet.")
    }

}

// Confirm betAmount amount available from bank

// Spin roulette wheel
function spinWheel() {
    spinBtn.removeEventListener("click", spinWheel);
    betDisplay_p.innerText = "No more bets.";
    bettingAllowed = false;

    console.log(`Bank balance:  ${bankBalance}`)
    bankBalance_span.innerText = bankBalance
    console.log("Spinning wheel...")
    // Get wheel result
    wheelResult = getWheelResult()
    // Display Results
    setTimeout(
        displayWheelResult,
        3000
    );
    setTimeout(resetTable, 6000)
}

function getWheelResult() {
    return Math.floor(Math.random() * table.wheel.length)
    console.log("got wheel result")
}

// Check for wins by payout (switch message?)
function checkForWins() {
    // Check for inside bets won

    // Check for outside bets won

    console.log("check for wins")
}

function manageBetting() {

    console.log("managed betting")
}

// Award bets
function payOutWins() {

    console.log("pay out wins")
}

function displayWheelResult() {
    var color
    if (table.red.includes(table.wheel[wheelResult])) {
        color = "Red";
    } else if (table.black.includes(table.wheel[wheelResult])) {
        color = "Black"
    } else color = "Green"
    messageDisplay_p.innerText = `${color} ${table.wheel[wheelResult]}`
    console.log(`${color} ${table.wheel[wheelResult]}`)
    console.log("display wheel result")
}

// Add result to list of results
// Remove oldest result from list of results
// Update results list display
// Display win or lost


// Reset table
function resetTable() {
    // Reset betAmount selection
    // Reset betAmount amount
    betAmount.value = 0;
    betDisplay_p.innerText = "Place your bet."
    // Re-enable betting
    betSelection = ""
    bettingAllowed = true;
    spinBtn.addEventListener("click", spinWheel);
    //updateBalance();
    console.log("reset table")
}


// updateBalance

table.js
// board.js

// American Roulette Wheel
export const wheel = [
    "0", "28", "9", "26", "30", "11", "7", "20", "32", "17", "5", "22", "34",
    "15", "3", "24", "36", "13", "1", "00", "27", "10", "25", "29", "12", "8",
    "19", "31", "18", "6", "21", "33", "16", "4", "23", "35", "14", "2"
]

/**
 * Inside Bets
 * - Straight up (35 to 1 payout) - any single number, including 0 and 00
 * - Split (17 to 1 payout) - any two adjoining numbers vertical or horizontal
 * - Basket (11 to 1 payout) - 0, 00, 1, 2, 3
 * - Street (11 to 1 payout) - any three numbers horizontal (1, 2, 3, or 4, 5, 6, etc.)
 * - Corner (8 to 1 payout) - any four adjoining numbers in a block (eg 17, 18, 20, 21)
 * - Six Line (5 to 1) - any six numbers from two rows (eg 28, 29, 30, 31, 32, 33)
 */

/**
 * Outside Bets - 2 to 1 payouts
 * - column1, column2, column3
 * - dozen1, dozen2, dozen3
 * 
 * Outside Bets - 1 to 1 payouts
 * - red, black
 * - even, odd
 * - low (1 through 18), high (19 through 36) 
 */
export const column1 = ["1", "4", "7", "10", "13", "16", "19", "22", "25", "28", "31", "34"]
export const column2 = ["2", "5", "8", "11", "14", "17", "20", "23", "26", "29", "32", "35"]
export const column3 = ["3", "6", "9", "12", "15", "18", "21", "24", "27", "30", "33", "36"]
export const green = ["0", "00"]

export const first12 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
export const second12 = [13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
export const third12 = [25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36]

export const red = ["1", "3", "5", "7", "9", "12", "14", "16", "18", "19", "21", "23", "25", "27", "30", "32", "34", "36"]
export const black = ["2", "4", "6", "8", "10", "11", "13", "15", "17", "20", "22", "24", "26", "28", "29", "31", "33", "35"]

export const even = [2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36]
export const odd = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35]

export const low = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]
export const high = [19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36]
